users/views.py

```diff
@@ -0,0 +1,188 @@
+                response["code"] = status.HTTP_200_OK
+                if not Token.objects.filter(user=user).exists():
+                    token = Token.objects.create(user=user)
+                else:
+                    token = Token.objects.get(user=user)
+                response["token"] = token.key
+        except KeyError:
+            response["message"] = "Bad Request"
+        except Http404:
+            response["message"] = "User does not exist"
+
+        return Response(response)
+
+
+class UserAPIView(APIView):
+    """
+    User Management API
+    post to add user
+    put to login user
+    patch to update user
+    delete to remove user
+    """
+
+    def post(self, request):
+        """
+        This function adds a user to db
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        response = {}
+        try:
+            name = request.POST["name"]
+            email = request.POST["email"].lower()
+            password = request.POST["password"]
+            validate_password(password)
+            validate_email(email)
+            unique_email = not User.objects.filter(email=email).exists()
+            valid_name = name not in ("", " ")
+            if valid_name and unique_email:
+                User.objects.create_user(email=email, password=password, name=name)
+                response["message"] = "Successfully created user."
+                response["code"] = status.HTTP_201_CREATED
+            else:
+                response["message"] = "Email already exists."
+                if not valid_name:
+                    response["message"] = "Name can not be blank."
+                response["code"] = status.HTTP_400_BAD_REQUEST
+        except KeyError:
+            response["message"] = "Required fields don't exist."
+            response["code"] = status.HTTP_400_BAD_REQUEST
+        except IntegrityError as integrity_error:
+            response["message"] = str(integrity_error)
+            response["code"] = status.HTTP_409_CONFLICT
+        except ValidationError as validation_error:
+            response["message"] = " ".join(validation_error)
+            response["code"] = status.HTTP_400_BAD_REQUEST
+
+        return Response(response)
+
+    @authentication_classes(TokenAuthentication)
+    @permission_classes(IsAuthenticated)
+    def patch(self, request):
+        """
+        This function updates data of existing user
+        Args:
+            request(HttpRequest): Value containing request data
+        Returns:
+            (Response): A json object containing message and code
+        """
+        try:
+            user = request.user
+            if "email" in request.POST:
+                email = request.POST["email"]
+                validate_email(email)
+                if not User.objects.filter(email=email).exists():
```

Comment:
What will happen if the email already exists? Is there any logic missing in this? e.g. if an email exists with some other user what error will be thrown?